pkg/storegateway/storegatewaypb/rpc.proto

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) The Thanos Authors.
+// Licensed under the Apache License 2.0.
+
+syntax = "proto3";
+package gatewaypb;
```

Comment:
What was the package in Thanos? To keep backward compatibility between Mimir versions, we need to keep the same package here.

Reply:
Keep in mind this is the protobuf package, **not** the go package, so it's fine if they don't match.

Reply:
The previous protobuf package was `thanos`

Reply:
I think testes are passing because our backward compatibility tests (`integration/backward_compatibility_test.go`) are not testing the compatibility with older store-gateways, but I think renaming the protobuf package will be a breaking change which we can't afford (if I remember correctly, we already went through it in the past).